src/containers/ListingPage/ListingPage.jsx

```diff
@@ -18,23 +18,60 @@ import ViewBar from './components/ViewBar';
 
 class ListingPage extends Component {
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      sortType: 'title_asc',
+      viewType: 'grid',
+    };
+    this.onViewChangeToList = this.onViewChangeToList.bind(this);
+    this.onViewChangeToGrid = this.onViewChangeToGrid.bind(this);
+    this.onSortChange = this.onSortChange.bind(this);
+    this.viewType = this.viewType.bind(this);
+    this.sortType = this.sortType.bind(this);
+  }
+
   componentWillMount() {
-    const { fetchListing, params: { listingId } } = this.props;
-    fetchListing(listingId);
+    const { fetchListingByFilter, params: { listingId } } = this.props;
+    fetchListingByFilter(listingId);
+  }
+
+  onViewChangeToList() {
+    this.setState({ viewType: 'list' });
+  }
+
+  onViewChangeToGrid() {
+    this.setState({ viewType: 'grid' });
+  }
+
+  onSortChange(event) {
+    this.setState({ sortType: event.target.value });
+  }
+
+  sortType() {
+    return this.state.sortType;
+  }
+
+  viewType() {
```

Comment:
Isteden for å lage en viewType funskjon, kan du ikke bare sende this.state.viewType nedover?